### DISCUSSION TITLE
Is there anyone who makes custom targets (Paid or Free), maybe on Fivr etc?

### DISCUSSION THREAD

**Discussion by dcan999** [iNavFlight/inav#9534] [General]
I always seem to want/need a few features that are just too niche to be added in regular updates or requests, and although I am trying, I don't have enough experience to do the programming myself. Are there people who offer this service? Things like getting certain Sbus telemetry data to read on the INAV OSD (RPM etc), changing ADC channels that aren't currently assignable (ADC5 and ADC6) etc...nothing that I think (in my limited knowledge programming) is too crazy , but above my capability.

**Comment by DzikuVx** [comment]
Well, that most probably depends on how much you are willing to pay or convince the dev that working on a feature you want is worth his private time :)

**Reply by dcan999** [reply]
I am willing to pay, and pay even more if I can be taught what is being done, to eventually do it myself, I just can't find anyone who will do such things. I tried to redefine the ADC 5 and 6 using a youtube walkthrough, but my targets never worked, and when I asked about it on Github nobody relied or helped, so I figured it might require payment.

**Reply by DzikuVx** [reply]
Open source is free for everybody. But this also means that nobody is forced to do anything. Like I mentioned, there are 3 options:
- someobody will just do it at one point
- nobody will ever do it
- you will convince someone it's worth doing it for free
- somebody pays someone :)

**Reply by dcan999** [reply]
How do I find these people who I can pay to discuss?

**Comment by sensei-hacker** [comment]
Contact me deepmagicbeginshere@gmail.com
Not promising that I'll do whatever, but I'm curious to hear details of what you want.

**Comment by dcan999** [comment]
So here is my project

I am building a turbine plane and using INAV for OSD and stabilization. The turbine comes with a telemetry box that outputs the following telemetry protocols: JR, Futaba, JETI, HoTT.     I use ELRS so I can't get telemetry down to my radio (CRSF protocol doesn't have any of the telemetry items I need like RPM, EGT etc anyways), so I want it shown on the OSD. I know INAV will accept telemetry via aspare UART and I see HoTT is even an option, so I want to be able to show the data from this telemetry box/ECU on the OSD (turbine RPM, running status, fuel remaining, EGT) etc. 

Not sure how hard this will be, maybe somebody can tell me if its easy, hard, or impossible :)

**Comment by dcan999** [comment]
Anyone?

**Comment by Scavanger** [comment]
@dcan999 
INAV can output telemetry data, i.e. from the perspective of the control system, the FC is a telemetry sensor.
The ECU itself acts as a telemetry sensor, but INAV cannot read data from telemetry sensors.

**Reply by dcan999** [reply]
Ok, so the telemetry pull down options in the Telemetry row of the ports tab, is just selecting what protocol the FC will OUTPUT to your radio? 

I guess I had hope that what I wanted was possible since we can input telemetry data from ESC's like RPM etc. 

So there is no way to input what the telemetry box I have is outputting, to be shown on OSD?

**Comment by Scavanger** [comment]
> Ok, so the telemetry pull down options in the Telemetry row of the ports tab, is just selecting what protocol the FC will OUTPUT to your radio?

Exactly

> I guess I had hope that what I wanted was possible since we can input telemetry data from ESC's like RPM etc.

 The ESC protocol is a separate protocol that has nothing to do with SBUS, HOTT etc.

> So there is no way to input what the telemetry box I have is outputting, to be shown on OSD?

Not at the moment. 
It would of course be possible to implement this. But I think this is a feature that would only be used very rarely, and would be a bit more complex to implement.

**Reply by dcan999** [reply]
I appreciate the replies. I guess I thought it might be simpler than it is, since in theory even the simple-ish radios that are supported by the telemetry box (JR, JETI, Futaba etc) are taking all that telemetry data and converting it to read on theradio screen. 

I would really like to be able to see the running state, EGT, and ML fuel used on the OSD though, I know not many people would use it (especially since some turbine manufacturers even ban their turbines from FPV use) which is why I mentioned paying somebody to do it, or guide me to do it, but it's feeling like devs would rather not.

**Comment by sensei-hacker** [comment]
> I use ELRS so I can't get telemetry down to my radio

For getting it down to your radio, you can use another ELRS module running in airport mode to carry whatever data you want. Uart in, uart out, Airport doesn't care what protocol.

You may want run that ELRS link on 900Mhz so there isn't interference with the 2.4 GHz control link.

**Reply by dcan999** [reply]
I am not sure I follow, I can control my plane using ELRS/CRSF, and using a seperate RX, make my Radiomaster running EdgeTX show futaba/JR/Graupner telemetry data?

**Reply by dcan999** [reply]
After doing a little reading, I see what you mean, but that will also require a second TX at my position, on a laptop or similar. I was looking to get the telemetry to my radio so I can set up voice callouts for fuel and EGT etc, but maybe I can do that with a laptop to, as a last resort, it will be better than nothing.

Thank you for the recommendation.

**Reply by Scavanger** [reply]
Would be theoretically possible, you would need a telemetry converter that converts sbus/hott/ ... into the Crossfire format, an Arduino nano, for example, could do this. So your ECU -> Arduino -> Elrs receiver.
Unfortunately there is nothing for this yet.
I am a freelance programmer, you can contact me, I can make you an offer. andreas.kanzler@outlook.com

**Reply by dcan999** [reply]
I will shoot you an email, thank you. Although from what I can tell, CRSF itself doesn't have any of the datapoints we would need to transmit (RPM, EGT etc). This seems like a very complicated thing I want, unfortunately. It's seeming like somehow getting the data into the FC and using INAV to display on the OSD might be just as easy (hard).

**Reply by OptimusTi** [reply]
For what you are trying to do you don't need crfs. You need something that reads your telemetry module and spit it out to inav's osd.
 You'll need to provide more info on that module.

**Reply by sensei-hacker** [reply]
Yep, crsf would just get in the way.
To get it down to the ground /  radio, you want a dumb serial link. Just a dumb wire, which will carry the Sbus from air to ground. Only without the wire.  A wireless wire - nothing protocol aware, no conversions. Like airport mode, or any other transparent serial RF link.

For adding a bunch of new elements to the INAV OSD - that gets a lot more complicated. Probably paying someone's rent for a month or so at that point.

**Reply by dcan999** [reply]
The telemetry module seems straight forward, it plugs into the ECU of the turbine and then outputs the telemetry data on any of the listed protocols it supports.

[0] = ""TempHigh"",
[1] = ""Trim Low"",
[2] = ""SetIdle!"",
[3] = ""Ready"",
[4]= ""Ignition"",
[5] = ""FuelRamp"",
[6] = ""Glow Test"",
[7] = ""Running"",
[8] = ""Stop"",
[9] = ""FlameOut"",
[10] = ""SpeedLow"",
[11] = ""Cooling"",
[12] = ""Ignitor Bad"",
[13] = ""Starter Bad"",
[14] = ""Weak Gas"",
[15] = ""Start On"",
[16] = ""User Off"",
[17] = ""Failsafe"",
[18] = ""Low RPM"",
[19] = ""Reset"",
[20] = ""Rx PwFail"",
[21] = ""Pre Heat"",
[22] = ""Battery!"",
[23] = ""Time Out"",
[24] = ""Overload"",
[25] = ""Ign.Fail"",
[26] = ""BurnerOn"",
[27] = ""Starting"",
[28] = ""SwitchOv"",
[29] = ""Cal.Pump"",
[30] = ""Pump Limit"",
[31] = ""NoEngine"",
[32] = ""PwrBoost"",
[33] = ""Run-Idle"",
[34] = ""Run-Max "",
[35] = ""Restart "","

 I guess for running state it just outputs a number and I found a list of the number vs meaning (which I guess a LUA would decode?, but I can't find a complete list of items. At least from looking at what JETI and Futaba radio users get down to their radios, it at least has Rxvolts, EGT, ml fuel used, running state, RPM.

I had no monetery number in mind really, I guess my budget would be around $500 to get working OSD elements but that might be too much work for that price. 

I have decided as well for a KISS approach, I can just mount the GSU (small touchscreen that connects to ECU, normally unplugged before takeoff) in the view of the FPV camera to be able to view all of the above data, minus fuel used, which is a big one though, obviously.

**Reply by OptimusTi** [reply]
MWOSD that? @ShikOfTheRa ?

**Comment by sensei-hacker** [comment]
> I will shoot you an email, thank you. Although from what I can tell, CRSF itself doesn't have any of the datapoints we would need to transmit (RPM, EGT etc).

Yes, CRSF telemetry has a limited number of telemetry sensors it can carry.  The TX module typically communicates with the radio over sbus, so sbus may be a better protocol to use. Some radios also have an Sbus port on the bottom of the radio.

**Comment by sensei-hacker** [comment]
> For what you are trying to do you don't need crfs. You need something that reads your telemetry module and spit it out to inav's osd. You'll need to provide more info on that module.
